platform-core-module/src/main/java/com/softicar/platform/core/module/user/password/reset/PasswordResetDiv.java

```diff
@@ -0,0 +1,214 @@
+package com.softicar.platform.core.module.user.password.reset;
+
+import com.softicar.platform.ajax.document.IAjaxDocument;
+import com.softicar.platform.common.core.interfaces.ITestMarker;
+import com.softicar.platform.core.module.AGCoreModuleInstance;
+import com.softicar.platform.core.module.CoreCssClasses;
+import com.softicar.platform.core.module.CoreI18n;
+import com.softicar.platform.core.module.CoreTestMarker;
+import com.softicar.platform.core.module.file.stored.AGStoredFile;
+import com.softicar.platform.core.module.file.stored.StoredFileResource;
+import com.softicar.platform.core.module.page.service.PageServiceDocumentBuilder;
+import com.softicar.platform.core.module.page.service.PageServiceFactory;
+import com.softicar.platform.core.module.user.password.policy.IPasswordPolicy;
+import com.softicar.platform.core.module.user.password.policy.SofticarPasswordPolicy;
+import com.softicar.platform.core.module.user.password.quality.UserPasswordQualityDiv;
+import com.softicar.platform.core.module.web.service.WebServiceUrlBuilder;
+import com.softicar.platform.db.core.transaction.DbTransaction;
+import com.softicar.platform.dom.document.CurrentDomDocument;
+import com.softicar.platform.dom.elements.DomDiv;
+import com.softicar.platform.dom.elements.DomForm;
+import com.softicar.platform.dom.elements.DomImage;
+import com.softicar.platform.dom.elements.DomPasswordInput;
+import com.softicar.platform.dom.elements.bar.DomActionBar;
+import com.softicar.platform.dom.elements.button.DomButton;
+import com.softicar.platform.dom.elements.dialog.DomModalAlertDialog;
+import com.softicar.platform.dom.elements.label.DomLabelGrid;
+import com.softicar.platform.dom.elements.message.DomMessageDiv;
+import com.softicar.platform.dom.elements.message.style.DomMessageType;
+import com.softicar.platform.dom.event.DomEventType;
+import com.softicar.platform.dom.event.IDomClickEventHandler;
+import com.softicar.platform.dom.event.IDomEvent;
+import com.softicar.platform.dom.input.AbstractDomInput;
+import com.softicar.platform.dom.styles.CssDisplay;
+import java.util.Optional;
+
+public class PasswordResetDiv extends DomDiv {
```

Comment:
I really wonder it we need this whole DIV at all. Wouldn't it suffice to simply generate a new password and send it per mail to the user? That's the way, how the admin does the password reset.

Reply:
I do strongly belive so, yes.
Every (good) password reset mechanism I know allows users to input their new password themselves, exactly like this form.